### PR TITLE
Plugin E2E: Add support for testing alert query

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

This PR makes it possible to write e2e tests that verifies that a backend data source plugin can return data frames in a format that is compatible with alerting. 

The process for testing alerting compatibility is as follows:
1. Use `alertRuleEditPage` fixture to open a new, empty alert rule edit page OR use the `gotoAlertRuleEditPage` fixture to open an already existing alert rule. 
2. Get query editor row, set data source and fill in the query
3. Click the `Preview` button to evaluate the alert query. If successful, the data source is compatible with alerting. 

This `AlertRuleEditPage` model follows the same pattern as all the other page models - it can be instantiated with or without an id. As mentioned above, it's exposed through `alertRuleEditPage` fixture (new alert rules) and `gotoAlertRuleEditPage` fixture (existing alert rules). I've also added a `readProvisionedAlertRule` that makes it possible to read a provisioned alert rule. 

In practice, I think this will be useful to check for regressions in backend data source plugins. In the past, it has happened that changes to the backend code and how the data frames are being constructed leads to breakages in alerting. Running e2e tests that checks for breakages should be able to catch these regressions. 

**Which issue(s) this PR fixes**:

Part of https://github.com/grafana/plugin-tools/issues/796

Will follow up with a docs PR later. 

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/plugin-e2e@1.4.0-canary.779.878465c.0
  # or 
  yarn add @grafana/plugin-e2e@1.4.0-canary.779.878465c.0
  ```
</details>

### CODE REVIEW

**Review by mckn — APPROVED** [review]
I think this looks great! 

I think the flows are a bit different if you are defining the alert rules as `Grafana managed` and `Datasource managed` alert rules. But I guess it should be enough to test the datasource via the `Grafana managed` flow.

**Comment by sunker** [comment]
@josmperez do you want to have a look at the docs?

**Review by josmperez — APPROVED** [review]
ignore approval (added it to wrong PR). i will review soon.

**Comment by sunker** [comment]
@josmperez can you take a look at the docs?

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/alert-queries.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  await page.waitForFunction(() => window.monaco);
+  await queryA.getByGrafanaSelector(selectors.components.CodeEditor.container).click();
+  await page.keyboard.insertText('topk(5, max(scrape_duration_seconds) by (job))');
+  await expect(alertRuleEditPage.evaluate()).toBeOK();
+});
+```
+
+### Evaluating a provisioned alert rule
+
+You can also use a provisioned alert rule to test that your data source is compatible with alerting.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/alert-queries.md:38]:
```suggestion
You can also use a provisioned alert rule to test that your data source is compatible with alerting. For example:
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/alert-queries.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+sidebar_position: 60
+---
+
+## Introduction
+
+For backend data source plugins that have alerting [enabled](../../tutorials/build-a-data-source-backend-plugin.md#enable-grafana-alerting), it should be possible to define alerts based on the data source queries. Before an alert can be saved, the conditions for an alert definition is evalauted by the alert engine to ensure that the response from the data source is shaped correctly. You can use the `alertRulePage` fixture to verify that alert rules can be created from the output of a query returned by the data source.
+
+### Evaluating a new alert rule
+
+The following example uses the `alertRulePage` fixture. With this fixture, the test starts in the page for adding a new alert rule. We then fill in the alert rule query and call the `evaluate` function. Evaluate clicks the `Preview` button which will trigger a call to the `eval` endpoint to evaluate that the response of the data source query can be used to create an alert. The `toBeOK` matcher is used to verify the evaluation was successful.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/alert-queries.md:22]:
```suggestion
The following example uses the `alertRulePage` fixture. With this fixture, the test starts in the page for adding a new alert rule. You then fill in the alert rule query and call the `evaluate` function. Evaluate clicks the `Preview` button which triggers a call to the `eval` endpoint to evaluate that the response of the data source query can be used to create an alert. The `toBeOK` matcher is used to verify that the evaluation was successful.
```

**Review by josmperez — APPROVED** [review]
Nice work. LGTM.

**Pull request merged**